sim/battle-stream.ts

```diff
@@ -41,12 +41,14 @@ function splitFirst(str: string, delimiter: string, limit: number = 1) {
 export class BattleStream extends Streams.ObjectReadWriteStream<string> {
 	readonly debug: boolean;
```

Comment:
Also why are these readonly? It makes sense that a consumer might want to change these.

Reply:
Because when I originally ported it I did a manual `prefer-readonly` because I don't like things to be mutable :).

Reply:
I personally think of `readonly` and `const` as "it would be a mistake to mutate these" and/or "it would break something if these were mutated", rather than just "nothing is currently mutating these".

Reply:
Yes, but I'm of the opinion that it's almost a mistake to ever mutate anything. I prefer to design all my code to be immutable, which is why for me 'it would break something if these were mutated' is my default.